polaris-react/src/components/Scrollable/Scrollable.tsx

```diff
@@ -36,218 +32,86 @@ export interface ScrollableProps extends React.HTMLProps<HTMLDivElement> {
   onScrolledToBottom?(): void;
 }
 
-interface State {
-  topShadow: boolean;
-  bottomShadow: boolean;
-  scrollPosition: number;
-  canScroll: boolean;
-}
-
-export class Scrollable extends Component<ScrollableProps, State> {
-  static ScrollTo = ScrollTo;
-  static forNode(node: HTMLElement): HTMLElement | Document {
-    const closestElement = node.closest(scrollable.selector);
-    return closestElement instanceof HTMLElement ? closestElement : document;
-  }
-
-  state: State = {
-    topShadow: false,
-    bottomShadow: false,
-    scrollPosition: 0,
-    canScroll: false,
-  };
-
-  private stickyManager = new StickyManager();
-
-  private scrollArea: HTMLElement | null = null;
-
-  private handleResize = debounce(
-    () => {
-      this.handleScroll();
-    },
-    50,
-    {trailing: true},
-  );
-
-  componentDidMount() {
-    if (this.scrollArea == null) {
-      return;
-    }
-    this.stickyManager.setContainer(this.scrollArea);
-    this.scrollArea.addEventListener('scroll', () => {
-      window.requestAnimationFrame(this.handleScroll);
-    });
-    window.addEventListener('resize', this.handleResize);
-    window.requestAnimationFrame(() => {
-      this.handleScroll();
-      if (this.props.hint) {
-        this.scrollHint();
-      }
-    });
-  }
-
-  componentWillUnmount() {
-    if (this.scrollArea == null) {
-      return;
+export function Scrollable({
+  children,
+  className,
+  horizontal,
+  vertical = true,
+  shadow,
+  hint,
+  focusable,
+  onScrolledToBottom,
+  ...rest
+}: ScrollableProps) {
+  const [topShadow, setTopShadow] = useState(false);
+  const [bottomShadow, setBottomShadow] = useState(false);
+  const stickyManager = useRef(new StickyManager());
```

Comment:
Great callout. I don't think there's much overhead since we don't pass the container here, but I'd be happy to update with lazy ref.

Reply:
Unfortunately this forces me to put `stickyManager` into the dependency array on line 88. Thought on disabling the lint rule for that? I'd probably prefer to just leave it in the array since its reference shouldn't change and won't actually trigger extra effects.